### PR TITLE
Feature:lesson2

### PR DESCRIPTION
**Pull request by kanakonagiri**
# [lesson2](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#2%E3%81%93%E3%81%AEdom%E3%82%92javascript%E3%81%A7%E3%81%A4%E3%81%8F%E3%82%8Ahtml%E5%86%85%E3%81%AEul%E3%81%AE%E4%B8%AD%E3%81%AB%E5%B7%AE%E3%81%97%E8%BE%BC%E3%82%93%E3%81%A7%E3%81%8F%E3%81%A0%E3%81%95%E3%81%84)
## [comment]
I created a DocumentFragment object and then inserted it.

### CODE REVIEW

File: lesson2/script.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const text = document.createTextNode('これです');
+
+    const frag = new DocumentFragment();
+    a.appendChild(img);
+    a.appendChild(text);
+    li.appendChild(a);
+    frag.appendChild(li);
+
+    //ulに挿入
+    document.getElementById('js-lists').appendChild(frag);
```
Comment (kenmori) [inline lesson2/script.js]: no need fragment. because It's not using in loop.

File: lesson2/script.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    const href = document.createAttribute('href');
+    href.value = "1.html";
+    a.setAttributeNode(href);
+
+    const img = document.createElement('img');
+    const src = document.createAttribute('src');
+    src.value = "bookmark.png";
+    img.setAttributeNode(src);
+    const alt = document.createAttribute('alt');
+    alt.value = "ブックマーク";
```
Comment (kenmori) [inline lesson2/script.js]: this is good. but, Do you know that  node  which created by createElement method, have attribute? It's accessible itself.

```js
img.alt = "ブックマーク"
```

File: lesson2/script.js (hunk 1 above)
Comment (kanakonagiri) [reply lesson2/script.js]: ok. I understand that fragment is used when using loop.
Rewritten without using fragment.
35bdbfe

File: lesson2/script.js (hunk 2 above)
Comment (kanakonagiri) [reply lesson2/script.js]: I understand that the node created by the createElement method has attributes, so I could write it simpler.
35bdbfe

**Comment by kenmori** [comment]
approve

**Pull request merged**